### PR TITLE
introduce readthedocs

### PR DESCRIPTION
**Pull request by gyorb**
with the config files the documentation can be viewed
on readthedocs.io

### CODE REVIEW

**Comment by gyorb** [comment]
With readthedocs we can generate and host the documentation. You can checkout a demo [here](https://codechecker-gyorb.readthedocs.io/en/readthedocs/).
What do you think?

**Comment by Xazax-hun** [comment]
LGTM, more exposure is always good. The question is, do we want to support this when the rendering differs from the github one?

**Comment by gyorb** [comment]
What rendering differences do you mean? I do not think we use the github flavored markdown in the documentation. Until we stick to simple markdown it shouldn't be a problem. If we want more complex documentation and markdown is not enough in the future we can switch to `rst`, in that case `Sphinx`  (supported by readthedocs) can be used to generate the documentation.
In the background mkdocs is called to generate the docs, you can do it locally too to view the results.

**Comment by Xazax-hun** [comment]
Sounds good.

File: readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: 2
```
Comment (csordasmarton) [inline readthedocs.yml]: Could you please add a target in the [Makefile](https://github.com/Ericsson/codechecker/blob/master/Makefile) which will build the documentation files (`build_mkdocs`)?

Comment (gyorb) [reply readthedocs.yml]: Yes but I would put that into a separate pull request because the two things are different.

Comment (gyorb) [reply readthedocs.yml]: See  #2026

**Pull request merged**